### PR TITLE
23833426 create user details page

### PR DESCRIPTION
**Pull request by naomiinanna**

### CODE REVIEW

File: src/components/UserDetailPage/UserDetailPage.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+function UserDetailPage() {
+
+  const [books, setBooks] = useState<Book[]>([]);
+
+  useEffect(() => {
+    fetch('https://localhost:8000/my/favorites')
+      .then(response => response.json())
+      .then(data => {
+        const res = data.data.map((book: Book) => {
```
Comment (tinkabel85) [inline src/components/UserDetailPage/UserDetailPage.tsx:15]: it does not work at my side. I think it is  data.favorites.map(....
- as per endpoint return value:
return $this->json(['favorites' => $user->getFavorites()]);

Comment (anna-bc) [reply src/components/UserDetailPage/UserDetailPage.tsx:15]: Also the token for authorization needs to be passed, too

File: src/components/UserDetailPage/UserDetailPage.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.UserDetailPage{
+    
+}
```
Comment (anna-bc) [inline src/components/UserDetailPage/UserDetailPage.scss:3]: no styling?

File: src/components/UserDetailPage/UserDetailPage.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          const authors = book.authors.map((author: Author) => `${author.lastName}`);
+          const categories = book.categories.map((category: Category) => `${category.categoryName}`);
+          return { ...book, authors, categories };
+        });
+        setBooks(res);
+      })
+      .catch(error => console.error(error));
+  }, []);
```
Comment (anna-bc) [inline src/components/UserDetailPage/UserDetailPage.tsx:37]: I'm not sure about this, but would it not be better to write this together in one useEffect? As both will be run on mounting...

**Pull request merged**